File: sig-ts/src/core/hash.ts
import { createHash } from 'node:crypto';

export function sha256(content: string): string {
  return createHash('sha256').update(content, 'utf8').digest('hex');
}

export function formatHash(hex: string): string {
  return `sha256:${hex}`;
}

export function parseHash(hash: string): { algorithm: string; hex: string } {
  const idx = hash.indexOf(':');
  if (idx === -1) {
    return { algorithm: 'sha256', hex: hash };
  }
  return { algorithm: hash.slice(0, idx), hex: hash.slice(idx + 1) };
}

File: sig-py/src/sig/core/hash.py
import hashlib


def sha256(content: str) -> str:
    """SHA-256 hash of UTF-8 encoded content, returns lowercase hex."""
    return hashlib.sha256(content.encode("utf-8")).hexdigest()


def format_hash(hex_str: str) -> str:
    return f"sha256:{hex_str}"


def parse_hash(hash_str: str) -> tuple[str, str]:
    """Returns (algorithm, hex). Defaults to sha256 if no prefix."""
    if ":" in hash_str:
        alg, hex_val = hash_str.split(":", 1)
        return alg, hex_val
    return "sha256", hash_str
